src/Tickets/Emails/Hooks.php

```diff
@@ -73,7 +73,9 @@ protected function add_filters() {
 	 *
 	 */
 	public function action_register_post_type() {
-		$this->container->make( Email_Handler::class )->register_post_type();
+		$email_handler = $this->container->make( Email_Handler::class );
+		$email_handler->register_post_type();
```

Comment:
This should be pretty much reserved for hooks. The logic should live in `Email_Handler`. We should probably have the `$this->maybe_populate_tec_tickets_emails_post_type` [here](https://github.com/the-events-calendar/event-tickets/pull/2550/files#diff-c97bfb6de59c73509e252cc9795314ed13c6b61585f3b735b84c66ca9a8a93ffR67)

Reply:
I tested with it there and it caused an error. It needed to be run on the `init` hook. I can separate it into a different method in Hooks if you think that would be better. Let me know your thoughts.

Reply:
I created a new method within the hooks file for this.